pageserver/src/metrics.rs

```diff
@@ -491,10 +491,94 @@ pub fn remove_tenant_metrics(tenant_id: &TenantId) {
 
 use futures::Future;
 use pin_project_lite::pin_project;
+use std::collections::HashMap;
 use std::pin::Pin;
+use std::sync::{Arc, Mutex};
 use std::task::{Context, Poll};
 use std::time::Instant;
 
+pub struct RemoteTimelineClientMetrics {
+    tenant_id: String,
+    timeline_id: String,
+    remote_operation_time: Mutex<HashMap<(&'static str, &'static str, &'static str), Histogram>>,
+    unfinished_tasks: Mutex<HashMap<(&'static str, &'static str), IntGauge>>,
+}
+
+impl RemoteTimelineClientMetrics {
+    pub fn new(tenant_id: &TenantId, timeline_id: &TimelineId) -> Self {
+        RemoteTimelineClientMetrics {
+            tenant_id: tenant_id.to_string(),
+            timeline_id: timeline_id.to_string(),
+            remote_operation_time: Mutex::new(HashMap::default()),
+            unfinished_tasks: Mutex::new(HashMap::default()),
+        }
+    }
+    pub fn remote_operation_time(
+        &self,
+        file_kind: &RemoteOpFileKind,
+        op_kind: &RemoteOpKind,
+        status: &'static str,
+    ) -> Histogram {
+        // XXX would be nice to have an upgradable RwLock
+        let mut guard = self.remote_operation_time.lock().unwrap();
+        let key = (file_kind.as_str(), op_kind.as_str(), status);
```

Comment:
Minor:

Is there any point to form the key out of `str` not the `Copy` enums that are already present (`status` could be also replaced with an enum)?
`(&'static str, &'static str, &'static str)` would become more self-documentable there (ideally, there could be a separate struct with names created)

Reply:
I'm hoping that in the future we can find some zero-cost way to curry `get_metric_with_label_values`. 
That would certainly be using `str`.
All of this code could be deleted, then, and I feel like it's more obvious to a future if the keys are  just `str` tuples.

See  https://neondb.slack.com/archives/C0277TKAJCA/p1671015002722739

Reply:
> zero-cost
> That would certainly be using `str`.

Not quite sure I agree, but ok, let's see how it gets transformed later.